samples_project/Assets/SampleViewer/Samples/VRSample/Scripts/ContinuousMovement.cs

```diff
@@ -1,25 +1,59 @@
-using System.Collections;
-using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.XR;
 using UnityEngine.XR.Interaction.Toolkit;
 using Unity.XR.CoreUtils;
+using UnityEngine.InputSystem;
+using UnityEngine.UI;
+using CommonUsages = UnityEngine.XR.CommonUsages;
 
 [RequireComponent(typeof(CharacterController))]
 public class ContinuousMovement : MonoBehaviour
 {
     public XRNode leftInputSource;
     public XRNode rightInputSource;
+    public InputAction LeftMenuAction;
+    public InputAction RightMenuAction;
+    public GameObject UICanvas;
+    public GameObject Instructions;
+    public bool UseSnapTurn;
+    public GameObject playerCamera;
+    public XRInteractorLineVisual leftLineVisual;
+    public XRInteractorLineVisual rightLineVisual;
+    public XRRayInteractor leftRayInteractor;
+    public XRRayInteractor rightRayInteractor;
     private Vector2 leftInputAxis;
     private Vector2 rightInputAxis;
     private CharacterController controller;
     [SerializeField] private float speed;
     [SerializeField] private float upSpeed;
+    private Toggle smoothTurnToggle;
     private XROrigin rig;
-    public LayerMask groundLayer;
     private float fallSpeed;
     private float heightOffset = 0.2f;
-    // Start is called before the first frame update
+    private bool toggledOn = true;
+    [SerializeField] private ContinuousTurnProviderBase smoothTurn;
+    [SerializeField] private SnapTurnProviderBase snapTurn;
+
+    private void Awake()
```

Comment:
It would probably be better to move these initialization calls to `Start()`, just to consolidate initialization in one place